symengine/parser.cpp

```diff
@@ -196,6 +208,31 @@ class ExpressionParser
                                  parse_string(iter + 1, operator_end[iter]));
                     iter = operator_end[iter] - 1;
 
+                } else if (s[iter] == '=') {
+                    result = Eq(result,
+                                parse_string(iter + 1, operator_end[iter]));
+                    iter = operator_end[iter] - 1;
+
+                } else if (s[iter] == '<' and s[iter + 1] == '=') {
```

Comment:
Did you try and look into this further? Why does adding `&& iter + 1 < end` cause it to throw everytime?